Source/WebCore/dom/QualifiedNameCache.cpp

```diff
@@ -65,21 +67,23 @@ Ref<QualifiedName::QualifiedNameImpl> QualifiedNameCache::getOrCreate(const Qual
     if (addResult.isNewEntry) {
         auto nodeNamespace = findNamespace(components.m_namespaceURI);
         auto elementName = findElementName(nodeNamespace, components.m_localName);
-
-        updateImplWithNamespaceAndElementName(impl, nodeNamespace, elementName);
+        auto attributeName = AttributeName::Unknown;
+        if (elementName == ElementName::Unknown)
+            attributeName = findAttributeName(nodeNamespace, components.m_localName);
```

Comment:
I can try that this in a follow-up. Cameron has a similar idea. Personally, I felt it would look a little odd to have these 2 things in the same enumeration. I also don't think we construct QualifiedNames very often since they are atomized/cached and we have static ones for all well-known ones.

Reply:
Following up in https://github.com/WebKit/WebKit/pull/12716 (Still waiting on A/B testing results to make sure it doesn't regress Speedometer).